jormungandr/src/fragment/pool.rs

```diff
@@ -68,6 +87,24 @@ impl Pools {
             .filter(|(_, exists_in_logs)| !exists_in_logs)
             .map(|(fragment, _)| fragment);
 
+        if let Some(mut persistent_log) = self.persistent_log.as_mut() {
+            for fragment in new_fragments.clone() {
+                let entry = PersistentFragmentLog {
+                    time: SecondsSinceUnixEpoch::now(),
+                    fragment,
+                };
+                // this must be sufficient: the PersistentFragmentLog format is using byte array
+                // for serialization so we do not expect any problems during deserialization
+                let codec = bincode::DefaultOptions::new().with_fixint_encoding();
+                if let Err(err) = codec.serialize_into(&mut persistent_log, &entry) {
+                    tracing::error!(err = %err, "failed to write persistent fragment log entry");
+                }
+                if let Err(err) = persistent_log.write_all("\n".as_bytes()) {
```

Comment:
This is now a Frankenstein format with binary payloads and newlines, how is it supposed to be parsed?
Change the entry encoding to use a length prefix.

Reply:
> Change the entry encoding to use a length prefix.

Is there a particular reason to do that? We should be fine without that actually.

Reply:
> This is now a Frankenstein format with binary payloads and newlines

Removed newlines, thanks for noting.

Reply:
If you can decode a stream of bincode payloads, fine then.

Reply:
I'm not sure if this is achieved by `.allow_trailing_bytes` and deserializing entries one after another from the same stream.

Reply:
You can use [`allow_trailing_bytes`](https://docs.rs/bincode/1.3.3/bincode/config/trait.Options.html#method.allow_trailing_bytes) and read the stream in a loop until reaching its end or meeting an invalid payload.